### ISSUE TITLE
Can I set the DHCP lease time via vagrant-libvirt?

### ISSUE THREAD

**Issue by johanneskastl** [vagrant-libvirt/vagrant-libvirt#1792]
**Is your feature request related to a problem? Please describe.**

TL;DR: Can I influence the DHCP leasetime for the network created by vagrant-libvirt?

When using vagrant-libvirt on e.g. a laptop, the VM's IP addresses might change if you put the laptop to sleep/suspend and wake it up again. Apparently, the DHCP lease time is only 1 hour,  so after that the IPs might change.

In my case it is not possible to set static IPs, so I would like to increase the DHCP lease time to e.g. 24h or even 1 week.

**Describe the solution you'd like**
Apparently one can set lots of things on private networks, but not for the public network.
https://vagrant-libvirt.github.io/vagrant-libvirt/configuration.html#private-network-options
https://vagrant-libvirt.github.io/vagrant-libvirt/configuration.html#public-network-options

Although there are some DHCP-related options for private networks, there is nothing regarding the leasetime.

A sample Vagrantfile without the network options might look like this:

```
Vagrant.configure("2") do |config|

  # name the VMs
  config.vm.define "leap155" do |node|

    # which image to use
    node.vm.box = "opensuse/Leap-15.5.x86_64"

    # disable synced folders
    node.vm.synced_folder ".", "/vagrant", disabled: true

    # sizing of the VMs
    node.vm.provider "libvirt" do |lv|
      lv.random_hostname = false
      lv.memory = 2048
      lv.cpus = 2
    end

    # set the hostname
    node.vm.hostname = "leap155"

  end # config.vm.define nodes
end
```

This results in a new network `vagrant-libvirt` being created whose xml looks like this:

```
network ipv6='yes'>                                                                                                                                                                         
  <name>vagrant-libvirt</name>                                                                                                                                                               
  <uuid>34d7cbbd-4cf4-4c4f-b9f6-8524f9101fc1</uuid>                                                                                                                                          
  <forward mode='nat'/>                                                                                                                                                                      
  <bridge name='virbr3' stp='on' delay='0'/>                                                                                                                                                 
  <mac address='52:54:00:77:6b:c8'/>                                                                                                                                                         
  <ip address='192.168.121.1' netmask='255.255.255.0'>                                                                                                                                       
    <dhcp>                                                                                                                                                                                   
      <range start='192.168.121.1' end='192.168.121.254'/>                                                                                                                                   
    </dhcp>                                                                                                                                                                                  
  </ip>                                                                                                                                                                                      
</network>
```

All it would need is one line inside the `<range>` according to [the libvirt documentation](https://libvirt.org/formatnetwork.html).

```
    <dhcp>
      <range start='192.168.121.1' end='192.168.121.254'/>
        <lease expiry='1' unit='hours'/>
      </range>
    </dhcp>
```

**Comment by abbbi** [comment]
its currently not implemented. I think it would be pretty straight forward to implement, the closest thing is to look at the old "bootp" related commits for what has to be changed:

both the network generation code and the template must be adopted:

```
lib/vagrant-libvirt/action/create_networks.rb
lib/vagrant-libvirt/templates/private_network.xml.erb
```

commit 80f8c6b65c0fa9f39b4fca3bf81b9fd38e90833f comes pretty close to this.

**Comment by johanneskastl** [comment]
@abbbi I tried to come up with something similar, see #1799.

This will only influence explicitly created private networks, I guess (if this even works...)
